services/storage/src/simcore_service_storage/simcore_s3_dsm.py

```diff
@@ -144,7 +150,16 @@ async def list_files(  # noqa C901
         data: list[FileMetaData] = []
         accessible_projects_ids = []
         async with self.engine.acquire() as conn, conn.begin():
-            accessible_projects_ids = await get_readable_project_ids(conn, user_id)
+            if project_id is not None:
```

Comment:
Can you maybe please document with a comment why is this if/else statement here? Thanks.

Reply:
Yep, good point. I added a line to the documentation of the function.